pkg/network/encoding/format.go

```diff
@@ -207,6 +207,7 @@ func httpKeyFromConn(c network.ConnectionStats) http.Key {
 func returnToPool(c *model.Connections) {
 	if c.Conns != nil {
 		for _, c := range c.Conns {
+			c.Reset()
```

Comment:
This will actually allocate and defeat the purpose of the `sync.Pool`. I think we want to simply set `HttpAggregations` to `nil` in `FormatConnection`.

Reply:
I'm not sure if I agree. 
True, there are other pointer fields whose objects we would "lose" (`Laddr`, `Raddr`, `IpTranslation` etc), but it appears that [actually _none_ of them is actually recycled](https://github.com/DataDog/datadog-agent/blob/dc2f20229531af78c0431093d9f2f8510a0586d2/pkg/network/encoding/format.go#L36-L74) ( in practice we're only pooling `agentmodel.Connection`) 

For HTTP stuff though, I agree that it doesn't matter whether we do `c.Reset()` or `HttpAggregations = nil`. I lean towards `c.Reset()` as a "catch-all" bugfix to prevent this from ever happening again in the future for other fields as well, but happy to change if you think otherwise.

Reply:
`c.Reset()` will create a new `model.Connection`, which is what the `sync.Pool` contains. That means it will allocate each time we serialize, which is what the pool is trying to avoid.

Reply:
I think you're wrong about that. 

I added a benchmark for it: https://github.com/DataDog/datadog-agent/pull/8510/commits/d3f818390e2c2ecde305ea093fea2beebd703b36

`
BenchmarkConnectionReset
BenchmarkConnectionReset-4   	293583999	         3.98 ns/op	       0 B/op	       0 allocs/op
`

Reply:
It appears there is a compiler optimization in play here, called `runtime.duffzero`, which is preventing the allocations from `*m = Connection{}`. It we are going to use `Reset()`, then we should get rid of the `sync.Pool`, and do some profiling to ensure this doesn't cause a perf regression (especially on Windows, since this was added during profiling work there #6709).

IMHO it would be safer and faster to set `HttpAggregations = nil` for now.

Reply:
honestly I don't understand why you say we should get rid of `sync.Pool` because we're calling `Reset()`.
I'm not convinced that this is compiler optimization either. I tried to run it with`gcflags=-N` and I'm getting the same result. Is there a way to validate what you're saying?
I might be missing something, but why would`*m = Connection{}` allocate? my understanding is that we're allocating an empty `Connection` on struct on the stack and then copying it's *value* to a (existing/previously allocated) memory region. I know that if the struct is too large
```
type  Foo struct {
b [1024]byte
```
Go could always allocate Foo directly into the heap, but I don't think this is happening here.
In any case, could you explain you reasoning? I'm just curious.
In any case I'll change `c.Reset()` to `HttpAggregations = nil` because I don't feel strongly about and they will achieve the same thing.

Reply:
> I don't understand why you say we should get rid of `sync.Pool` because we're calling Reset()

Yeah, this is my bad. I was thinking the pointer value was changing, so we would be putting back in a different object, but that isn't what it is happening.

> Is there a way to validate what you're saying?

I discovered `runtime.duffzero` when capturing a CPU profile while running your benchmark. I suppose that it could be used to zero-initialize the struct in `Reset()`. I wouldn't be surprised if the compiler recognized this pattern and simplified it to simply zero-init the destination. It is hard to find good documentation on the compiler optimizations since they aren't supposed to be relied upon.

> I might be missing something, but why would `*m = Connection{}` allocate?

This is an allocation. It may be on the stack, rather than the heap, but an allocation regardless. The Go language spec doesn't actually mention stack or heap, since it does escape analysis to determine where to allocate.

I think I avoided using `Reset()` before, because it felt unnecessary when we should be setting every field on serialization. Using `Reset()` is definitely more defensive, especially once we start having more optional fields. It does result in more operations, but unlikely to be noticeable in any profiling.